clang/lib/Sema/SemaInit.cpp

```diff
@@ -8192,6 +8192,14 @@ ExprResult InitializationSequence::Perform(Sema &S,
             Kind.getRange().getEnd());
       } else {
         CurInit = new (S.Context) ImplicitValueInitExpr(Step->Type);
+        // Note the return value isn't used to return early
+        // to preserve the AST as best as possible even though an error
+        // might have occurred. For struct initialization it also allows
+        // all field assignments to be checked rather than bailing on the
+        // first error.
+        S.BoundsSafetyCheckInitialization(Entity, Kind, Sema::AA_Initializing,
```

Comment:
Not sure we really need this comment as that much is pretty much implied by discarding the return value (this comment is also duplicated somewhere else, which isn’t exactly ideal). I think we can just remove this?

Reply:
The comment is explaining **why** we aren't checking the return value. If you remove it isn't obvious at all why its being ignored.